src/WebJobs.Script.WebHost/Management/VirtualFileSystem.cs

```diff
@@ -641,10 +640,43 @@ private IEnumerable<VfsStatEntry> GetDirectoryResponse(HttpRequest request, File
         protected HttpResponseMessage CreateResponse(HttpStatusCode statusCode, object payload = null)
         {
             var response = new HttpResponseMessage(statusCode);
-            if (payload != null)
+            try
+            {
+                if (payload != null)
+                {
+                    // Use safe serialization settings
+                    var jsonSerializerSettings = new JsonSerializerSettings
+                    {
+                        NullValueHandling = NullValueHandling.Ignore,
+                        DefaultValueHandling = DefaultValueHandling.Include,
+                        Formatting = Formatting.None
+                    };
+
+                    // Check if the payload is a string or an exception
+                    payload = payload is string ? payload as string : payload is Exception ? (payload as Exception).Message : payload;
+
+                    // Sanitize the payload if it's an object
+                    var content = payload is string ? payload as string : JsonConvert.SerializeObject(payload, jsonSerializerSettings);
+                    response.Content = new StringContent(content, Encoding.UTF8, "application/json");
+                }
+            }
+            catch (JsonSerializationException je)
             {
-                var content = payload is string ? payload as string : JsonConvert.SerializeObject(payload);
-                response.Content = new StringContent(content, Encoding.UTF8, "application/json");
+                // Return a generic error message to avoid exposing sensitive details
```

Comment:
Previously for such exception cases, customers will get `JsonSerializationException` (with complete stack trace?) and no response. Now, customers will get InternalServerError in Http response. This looks change in API behavior to me if they depend on getting the exception instead of response. @jviau what are your thoughts on this?

Reply:
Any uncaught exception in AspNetCore will result in an `InternalServerError` so the status code isn't changing. The body of the response is changing though.

However, I think there may be some deeper discussion to be had with the error body. If we are changing the error body, what we change to is important. A service should have consistent error formats across its APIs. We should review if we already have a defined format and use that. If we don't have one we should discuss if defining one and moving this API to it is in scope.

Reply:
@surgupta-msft @jviau Let me know what is the final take here? If required, we can hop into a quick call

Reply:
@jviau I just wanted to let you know. Status code remains same, and we are just removing the exception trace from the response. Remaining should work as it was working already.